sunxspex/emission.py

```diff
@@ -348,6 +444,172 @@ def brm_guass_legendre(x1, x2, npoints):
     return x, w
 
 
+def brm2_dmlin(a, b, maxfcn, rerr, eph, eelow, eebrk, eehigh, p, q, z, efd=True, verbose=False):
+    """
+    This is used for thin-target calculation from a double power-law electron density distribution
+    To integrate a function via the method of Gaussian quadrature.  Repeatedly
+    doubles the number of points evaluated until convergence, specified by the
+    input rerr, is obtained, or the maximum number of points, specified by the
+    input maxfcn, is reached.  If integral convergence is not achieved, this
+    function returns the error code ier = 1 when either the maximum number of
+    function evaluations is performed or the number of Gaussian points to be
+    evaluated exceeds maxfcn.  Maxfcn should be less than or equal to 2^nlim,
+    or 4096 with nlim = 12.
+
+    This function splits the numerical integration into up to three parts and returns the sum of
+    the parts. This avoids numerical problems with discontinuities in the electron distribution
+    function at eelow and eebrk.
+
+    Parameters
+    ----------
+    a : np.array
+        Array containing lower integration limits
+    b : np.array
+        Array containing upper integration limits
+    maxfcn : int
+        Maximum number of points used in Gaussian quadrature integration
+    rerr : float
+        Desired relative error for integral evaluation. For example, rerr = 0.01 indicates that
+        the estimate of the integral is to be correct to one digit, whereas rerr = 0.001
+        alls for two digits to be correct.
+    eph : np.array
+        Array of photon energies at which to calculate fluxes
+    eelow : float
+        Low energy cut off of electron energy distribution, see broken_powerlaw()
+    eebrk : float
+        Break energy of electron energy distribution, see broken_powerlaw()
+    eehigh : float
+        High energy cut off of electron energy distribution, see broken_powerlaw()
+    p : float
+        Slope below the break energy
+    q : flaot
+        Slope above the break energy
+    z : float
+        Mean atomic number of plasma
+    efd: True (default) - electron flux density distribution, False - electron density distribution.
+         This input is not used in the main routine, but is passed to Brm_Fthin()
+    verbose: bool. Default False
+         If True, Print extra information in command line
+
+    Returns
+    -------
+    tuple : (Dmlin, irer)
+
+        Array of integral evaluation (Dmlin) and array of error flags
+        (ier, 0 for converged and 1 for nonconverge)
+
+    History:
+    -----
+    03/15/2020, Bin Chen (binchen@njit.edu). Initial version modified from SSW Brm2_Dmlin.pro:
+    https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_dmlin.pro
+
+    """
+    mc2 = 510.98
```

Comment:
Centralised units as in thicktarget function below

Reply:
done